### ISSUE TITLE
Display "No Posts" Message

### ISSUE THREAD

**Issue by coooolers** [sysgears/apollo-universal-starter-kit#693]
If there are no posts to display, the system should display a message "No Posts in the db"

I think that the existing system will throw an "unhandled" error if there are no existing posts in the database and the system tries to display the list of the existing posts.

**Comment by axelmm** [comment]
> I think that the existing system will throw an "unhandled" error if there are no existing posts in the database and the system tries to display the list of the existing posts.

This is not true, empty table is displayed.

```javascript
      return (
        <PageLayout>
          {this.renderMetaData()}
          <h2>{t('list.subTitle')}</h2>
          <Link to="/post/new">
            <Button color="primary">{t('list.btn.add')}</Button>
          </Link>
          <h1 />
          {!posts.edges.length &&
            <Alert color="warning" className="alert-noitems">
              {t('post.noItemsMsg')}
            </Alert>
          } 
          {!!posts.edges.length &&
            <React.Fragment>
              <Table dataSource={posts.edges.map(({ node }) => node)} columns={columns} />
              <Pagination
                itemsPerPage={posts.edges.length}
                handlePageChange={this.handlePageChange}
                hasNextPage={posts.pageInfo.hasNextPage}
                pagination={type}
                total={posts.totalCount}
                loadMoreText={t('list.btn.more')}
                defaultPageSize={itemsNumber}
              />
            </React.Fragment>
          }
        </PageLayout>
      );
```
This is simplest solution, going further you can create renderItems() and early return <Alert...   and the rest otherwise:

```javascript
  renderPosts = () => {
    const { posts, t } = this.props;    
    if( !posts.edges.length ) return (
        <Alert color="warning">
          {t('post.noItemsMsg')}
        </Alert>
    );               
    const columns = [
      {
        title: t('list.column.title'),
        dataIndex: 'title',
        key: 'title',
        render: (text, record) => (
          <Link className="post-link" to={`/post/${record.id}`}>
            {text}
          </Link>
        )
      },
      {
        title: t('list.column.actions'),
        key: 'actions',
        width: 50,
        render: (text, record) => (
          <Button
            color="primary"
            size="sm"
            className="delete-button"
            onClick={() => this.handleDeletePost(record.id)}
          >
            {t('post.btn.del')}
          </Button>
        )
      }
    ];
    return (
      <React.Fragment>
        <Table dataSource={posts.edges.map(({ node }) => node)} columns={columns} />
        <Pagination
          itemsPerPage={posts.edges.length}
          handlePageChange={this.handlePageChange}
          hasNextPage={posts.pageInfo.hasNextPage}
          pagination={type}
          total={posts.totalCount}
          loadMoreText={t('list.btn.more')}
          defaultPageSize={itemsNumber}
        />
      </React.Fragment>
    );
  }
  

  render() {
    const { loading, posts, t } = this.props;
    if (loading && !posts) {
      return (
        <PageLayout>
          {this.renderMetaData()}
          <div className="text-center">{t('post.loadMsg')}</div>
        </PageLayout>
      );
    } else {
      return (
        <PageLayout>
          {this.renderMetaData()}
          <h2>{t('list.subTitle')}</h2>
          <Link to="/post/new">
            <Button color="primary">{t('list.btn.add')}</Button>
          </Link>
          <h1 />
          {this.renderPosts()}
        </PageLayout>
      );
    }
  }
}
```


It simplifies main render() end defines columns only when needed.

**Comment by lyzhovnik** [comment]
Hi @coooolers , thank for your issue. We have fixed this in the #730 pull request.
